### PR TITLE
Revendor gardener/gardener after merge of gardener/gardener-extensions

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
Revendor gardener/gardener after merge of gardener/gardener-extensions

**Special notes for your reviewer**:
NONE

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: hack/generate-code [hunk 1]
```diff
@@ -24,7 +24,7 @@ bash "${PROJECT_ROOT}"/vendor/k8s.io/code-generator/generate-internal-groups.sh
   github.com/gardener/gardener-extension-provider-aws/pkg/apis \
```
Comment (ialidzhikov) [inline hack/generate-code:24]: You could add after the license header

```
set -o errexit
set -o nounset
set -o pipefail
```

File: Makefile [hunk 2]
```diff
@@ -34,70 +34,9 @@ ifeq ($(WEBHOOK_CONFIG_MODE), service)
...
-.PHONY: install-requirements
-install-requirements:
-	@go install -mod=vendor $(REPO_ROOT)/vendor/github.com/ahmetb/gen-crd-api-reference-docs
-	@go install -mod=vendor $(REPO_ROOT)/vendor/github.com/gobuffalo/packr/v2/packr2
-	@go install -mod=vendor $(REPO_ROOT)/vendor/github.com/golang/mock/mockgen
-	@go install -mod=vendor $(REPO_ROOT)/vendor/github.com/onsi/ginkgo/ginkgo
-	@$(REPO_ROOT)/vendor/github.com/gardener/gardener-extensions/hack/install-requirements.sh
-
-.PHONY: all
-ifeq ($(VERIFY),true)
```
Comment (ialidzhikov) [inline Makefile:76]: You could remove the VERIFY arg on top of this file as it seems unused.

File: vendor/github.com/gardener/gardener/hack/clean.sh [hunk 3]
```diff
@@ -18,8 +18,7 @@ set -e
 
 echo "> Clean"
 
-for source_tree in ${SOURCE_TREES[@]}; do
-  find "$(dirname "$source_tree")" -type f -name "controller-registration.yaml" -exec rm '{}' \;
```
[the remark this answers is not in the document]
Comment (rfranzke) [reply vendor/github.com/gardener/gardener/hack/clean.sh:22]: Nice catch, opened https://github.com/gardener/gardener/pull/2152 - will revendor it after merge.

**Comment by rfranzke** [comment]
Thanks @ialidzhikov - how is it now?

**Pull request merged**